import { AxiosError } from 'axios';
import { HandledError } from '../../HandledError';
export interface GithubV4Response<DataResponse> {
    data: DataResponse;
    errors?: {
        type: string;
        path: string[];
        locations: {
            line: number;
            column: number;
        }[];
        message: string;
    }[];
}
export declare function apiRequestV4<DataResponse>({ githubApiBaseUrlV4, accessToken, query, variables, handleError, }: {
    githubApiBaseUrlV4: string;
    accessToken: string;
    query: string;
    variables?: {
        [key: string]: string | number | null;
    };
    handleError?: boolean;
}): Promise<DataResponse>;
export declare function handleGithubV4Error(e: AxiosError<GithubV4Response<null>>): HandledError;
